src/common/host.rs

```diff
@@ -28,16 +28,15 @@ impl ::Header for Host {
         values
             .next()
             .cloned()
-            .map(Bytes::from)
-            .and_then(|bytes| Authority::from_shared(bytes).ok())
+            .map(|val| Bytes::copy_from_slice(val.as_bytes()))
```

Comment:
Since doing a copy here, perhaps we can skip the `copy_from_slice` step and just use `Authority::try_from(val.as_bytes())`?